src/logging_vnode.erl

```diff
@@ -384,19 +387,30 @@ handle_command({append, LogId, LogOperation, Sync}, _Sender,
 			true ->
 			    case disk_log:sync(Log) of
 				ok ->
-				    {reply, {ok, OpId}, State};
+					{{ok, OpId}, State};
 				{error, Reason} ->
-				    {reply, {error, Reason}, State}
+				    {{error, Reason}, State}
 			    end;
 			false ->
-			    {reply, {ok, OpId}, State}
+				{{ok, OpId}, State}
 		    end;
                 {error, Reason} ->
-                    {reply, {error, Reason}, State}
+	                {{error, Reason}, State}
             end;
         {error, Reason} ->
-            {reply, {error, Reason}, State}
-    end;
+	        {{error, Reason}, State}
+	end,
+	%% The following is used when this function is called by async_append,
+	%% to reply to the process calling that function.
+	%% This is currently being used by the update_objects function in the
+	%% interactive coordinator fsm.
+	case ReplyTo of
+		ignore->
+			{reply, Reply, NewState};
+		_->
+			gen_fsm:send_event(ReplyTo, Reply),
+			{noreply, NewState}
+	end;
```

Comment:
I think this additional ReplyTo is unnecessary. Just {reply, Reply, NewState} and using command/sync_command for async/sync calls gives you the desired behaviour.